### PR TITLE
Introduce app-serverless-cms and simplify Admin app setup

### PR DESCRIPTION
**Pull request by Pavel910**
## Changes
This PR introduces several major improvements to the Admin app:
- the whole Webiny admin app setup is now managed by a single `@webiny/app-serverless-cms` package
- the system is now powered by a new underlying framework, which makes it possible to compose/override any React component (if the composition is enabled on the component)
- the new API is designed with pure React in mind, meaning, it works using plain HOCs, React Contexts, and hooks
- the old plugins still work, until we transition the whole system to the React-styled API (but that will not happen over night)
- the amount of files in user's projects is greatly reduced, to make management and maintenance easier
- registration of extensions is now location agnostic: you don't need to register plugins into a global plugins registry. Instead, you simply mount components _anywhere_ and at any point in time (this enables you to code-split your extensions, mount them conditionally based on permissions or some business-logic)
- encapsulation of extensions is now possible: you only need 1 React component, and with that, you can add new React Context providers, compose components, add menus, routes, etc. without tinkering with the structure of the original application (Webiny CMS)
- since the new framework is based on React contexts, it doesn't enforce component hierarchy. This means that you can encapsulate and group your components however you see fit

Project file structure:
image

An example of creating new extensions (routes, menus, permission checks):
```tsx
import React from "react";
import { Extensions, Layout, AddMenu, AddRoute } from "@webiny/app-admin";
import { HasPermission } from "@webiny/app-security";
import FileManagerSettings from "~/admin/views/FileManagerSettings";

export const FileManager = () => {
    return (
        <Extensions>
            <HasPermission name={"fm.settings"}>
                <AddRoute path="/settings/file-manager/general">
                    <Layout title={"File Manager Settings - General"}>
                        <FileManagerSettings />
                    </Layout>
                </AddRoute>
                <AddMenu id={"settings"}>
                    <AddMenu id={"settings.fileManager"} label={"File Manager"}>
                        <AddMenu
                            id={"settings.fileManager.general"}
                            label={"General"}
                            path={"/settings/file-manager/general"}
                        />
                    </AddMenu>
                </AddMenu>
            </HasPermission>
        </Extensions>
    );
};
```

## How Has This Been Tested?
Manually and using E2E tests.

## Documentation
https://deploy-preview-371--webiny-docs.netlify.app/docs/how-to-guides/webiny-applications/admin-area/introduction/
https://deploy-preview-371--webiny-docs.netlify.app/docs/how-to-guides/webiny-applications/admin-area/framework/

### CODE REVIEW

File: packages/app-admin/package.json [hunk 1]
```diff
@@ -18,14 +19,16 @@
...
-    "@webiny/plugins": "^5.20.0",
-    "@webiny/react-router": "^5.20.0",
-    "@webiny/ui": "^5.20.0",
-    "@webiny/ui-composer": "^5.20.0",
-    "@webiny/validation": "^5.20.0",
+    "@webiny/app": "^5.18.3",
+    "@webiny/app-admin-core": "^5.18.3",
+    "@webiny/app-security": "^5.18.3",
+    "@webiny/form": "^5.18.3",
+    "@webiny/plugins": "^5.18.3",
```
Comment (adrians5j) [inline packages/app-admin/package.json:26]: Maybe we should update these.

File: packages/app-admin/src/base/Base.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import { ReactComponent as SlackIcon } from "~/assets/icons/slack-logo.svg";
+import { ReactComponent as GithubIcon } from "~/assets/icons/github-brands.svg";
+import { ReactComponent as FileIcon } from "~/assets/icons/insert_drive_file-24px.svg";
+import { ReactComponent as SettingsIcon } from "~/assets/icons/round-settings-24px.svg";
+import { FileManager } from "~/components";
+
+import adminPlugins from "../plugins";
+
+const BaseExtension = () => {
+    plugins.register(adminPlugins());
```
Comment (adrians5j) [inline packages/app-admin/src/base/Base.tsx:15]: This is backwards compatibility-related?

File: packages/app-graphql-playground/package.json [hunk 3]
```diff
@@ -11,13 +11,12 @@
...
     "@emotion/styled": "^10.0.27",
-    "@webiny/app": "^5.20.0",
-    "@webiny/app-admin": "^5.20.0",
-    "@webiny/app-i18n": "^5.20.0",
-    "@webiny/app-security": "^5.20.0",
-    "@webiny/plugins": "^5.20.0",
-    "@webiny/react-router": "^5.20.0",
-    "@webiny/ui": "^5.20.0",
+    "@webiny/app": "^5.18.3",
+    "@webiny/app-admin": "^5.18.3",
```
Comment (doitadrian) [inline packages/app-graphql-playground/package.json:15]: Old 5.18.3 versions here.

**Review by adrians5j — COMMENTED** [review]
Can't say anything constructive except huge PR 🤯 

BTW does this branch include the Webpack5/TS upgrades @kedrzu was doing?

File: packages/app-admin/src/base/Base.tsx (hunk 2 above)
Comment (Pavel910) [reply packages/app-admin/src/base/Base.tsx:15]: Yes, we'll be using this technique to hide plugins from user's projects, but still use them internally.

File: packages/app-admin/package.json (hunk 1 above)
Comment (Pavel910) [reply packages/app-admin/package.json:26]: > BTW does this branch include the Webpack5/TS upgrades @kedrzu was doing?

No, it's a separate thing.

**Comment by Pavel910** [comment]
Closing in favor of https://github.com/webiny/webiny-js/pull/2165

**Pull request closed without merging**